crates/wallet/src/descriptor/policy.rs

```diff
@@ -644,6 +640,57 @@ impl Policy {
         Ok(Some(policy))
     }
 
+    fn make_multi_a<Ctx: ScriptContext + 'static>(
```

Comment:
Thanks for the attempt to rewrite what is here! I would not bother though, since we want to replace this `policy` stuff with `miniscript`'s planning module.

Reply:
Now I understand. The only difference is when constructing the `Threshold`, we enforce different max `n` (of the `k-of-on` theshold) for different script constructions.

Can we use a const generic instead of duplicating code?

```rust
    fn make_multi<Ctx: ScriptContext + 'static, const MAX: usize>(
        threshold: &Threshold<DescriptorPublicKey, MAX>,
        signers: &SignersContainer,
        build_sat: BuildSatisfaction,
        sorted: bool,
        secp: &SecpCtx,
    ) -> Result<Option<Policy>, PolicyError> { todo!() }
```

Reply:
Due to time constraints I'm going to do this and push a commit on top.

Reply:
Done in 11200810d048abb8deb3c7961ca93c68011d41fd

Reply:
Great! That's much better, thanks for the suggestion and for addressing this @evanlinjin!